moveit_plugins/moveit_simple_controller_manager/src/moveit_simple_controller_manager.cpp

```diff
@@ -44,6 +44,23 @@
 #include <algorithm>
 #include <map>
 
+namespace
+{
+/**
+ * @brief Compose a parameter name used to load controller configuration from a ROS parameter.
+ * @param base_namespace Namespace in which to look up the parameter.
+ * @param controller_name Name of the controller configured by this parameter.
+ * @param param_name Specific name of the parameter.
+ * @return std::string concatenating the parameters separated by the `.` character. For example:
+ * base_namespace.controller_name.param_name
+ */
+std::string makeControllerParameterName(const std::string& base_namespace, const std::string& controller_name,
+                                        const std::string& param_name)
+{
+  return std::string(base_namespace).append(".").append(controller_name).append(".").append(param_name);
+}
+}  // namespace
```

Comment:
Thank you for adding this and fixing the issue with strings and clang-tidy.  Long term I hope to make it easier to have subsystems of the node use parameters and that be really straightforward.  There are a couple of PRs to rclcpp you might find interesting relating to this:

https://github.com/ros2/rclcpp/pull/1824
https://github.com/ros2/rclcpp/pull/1796